c2rust-refactor/src/scripting/to_lua_ast_node.rs

```diff
@@ -462,3 +539,83 @@ impl ToLuaExt for AstNode {
         }
     }
 }
+
+/// FnLike AST node handle
+//
+// This object is NOT thread-safe. Do not use an object of this class from a
+// thread that did not acquire it.
+// @type FnLikeAstNode
+unsafe impl Send for LuaAstNode<P<FnLike>> {}
+impl UserData for LuaAstNode<FnLike> {
+    fn add_methods<'lua, M: UserDataMethods<'lua, Self>>(methods: &mut M) {
+        methods.add_method("get_kind", |lua_ctx, this, ()| {
+            Ok(this.0.borrow().kind.to_lua(lua_ctx))
```

Comment:
Just return the name with `AstName`, as noted above.

Reply:
I don't think this is supported for `FnLike` since that's a custom struct of ours. Should I give it a manual impl?

Reply:
Hm, good point. Yeah, a manual impl is more generic, and can just delegate to `AstName` for the contained type.